src/lsdb/loaders/dataframe/from_dataframe.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+import math
+
+import pandas as pd
+
+from lsdb.catalog import Catalog
+from lsdb.loaders.dataframe.dataframe_catalog_loader import DataframeCatalogLoader
+
+
+def from_dataframe(
+        df: pd.DataFrame,
+        lowest_order: int = 0,
+        highest_order: int = 10,
+        partition_size: float | None = None,
+        threshold: int | None = None,
+        **kwargs,
+) -> Catalog:
+    """Load a catalog from a Pandas Dataframe in CSV format.
+
+    Args:
+        df (pd.Dataframe): The catalog Pandas Dataframe
+        lowest_order (int): The lowest partition order
+        highest_order (int): The highest partition order
+        partition_size (float): The desired partition size, in megabytes
+        threshold (int): The maximum number of data points per pixel
+        **kwargs: Arguments to pass to the creation of the catalog info
+
+    Returns:
+        Catalog object loaded from the given parameters
+    """
+    if threshold is not None and partition_size is not None:
+        raise ValueError("Specify only one: threshold or partition_size")
+    if threshold is None and partition_size is None:
+        threshold = DataframeCatalogLoader.DEFAULT_THRESHOLD
+    elif threshold is None and partition_size is not None:
+        threshold = calculate_threshold(df, partition_size)
+    assert isinstance(threshold, int)
```

Comment:
I think this logic should be in the loader. So that if you want to use the loader elsewhere it'll be more feature complete